### PR TITLE
Speed up Encoder and MelEncoder

### PR DESCRIPTION
**Pull request by ilya16**
Both `Encoder` and `MelEncoder` work individually with batched inputs to provide correct computation of convolution and normalization on padded variable length inputs. Example:
https://github.com/NVIDIA/flowtron/blob/f79b9806256dc5ddd342c0c571c20c87a67c839a/flowtron.py#L338-L344

This PR adds `MaskedInstanceNorm1d` (`InstanceNorm1d` accepting optional input mask) and rewrites `Encoder` and `MelEncoder` forward passes without any loops. `MaskedInstanceNorm1d` is compatible with `InstanceNorm1d`.

The correctness of computation is satisfied by masking out padded values before applying convolution and computing Instance Normalization statistics only for non-padded positions inside `MaskedInstanceNorm1d`. During inference, the module is equivalent to `InstanceNorm1d`.

Training Flowtron with this module does not degrade the results and slightly speeds up the computation.

<details>
  <summary>Proof of concept for <code>MaskedInstanceNorm1d</code>:</summary>

  ```python
import torch
import torch.nn as nn
from flowtron import get_mask_from_lengths, MaskedInstanceNorm1d

N, C, L = 8, 80, 100
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')

# random tensor
x = torch.randn((N, C, L), device=device)
x1 = x.clone().detach()

# input lengths and mask
lens = torch.randint(L // 4, L, size=(N,), device=device)
lens[0] = L
mask = get_mask_from_lengths(lens).unsqueeze(1)

# standard InstanceNorm1d and old Encoder-like code
in_pt = nn.InstanceNorm1d(C, affine=True).to(device)

x_embedded = []
for i in range(x.size(0)):
    curr_x = x[i:i+1, :, :lens[i]].clone()
    curr_x = in_pt(curr_x)
    x_embedded.append(curr_x[0].transpose(0, 1))
out_in_pt = torch.nn.utils.rnn.pad_sequence(x_embedded, batch_first=True)

# MaskedInstanceNorm1d and optimized computation
in_m = MaskedInstanceNorm1d(C, affine=True).to(device)

out_in_m = in_m(x1, mask)
out_in_m.masked_fill_(~mask, 0.)  # mask padding positions that got affected by normalization
out_in_m = out_in_m.transpose(1, 2)

# compare outputs
print(torch.abs(out_in_pt - out_in_m).mean())
# tensor(2.6331e-08, device='cuda:0')
  ```

Reference: [Gist](https://gist.github.com/ilya16/c622461000480e66ae906dd9dbe8ea26) with other masked normalization layers: `MaskedBatchNorm{1d,2d,3d}`, `MaskedInstanceNorm{1d,2d,3d}`.
</details>

### CODE REVIEW

**Comment by rafaelvalle** [comment]
Thank you for the pr!
